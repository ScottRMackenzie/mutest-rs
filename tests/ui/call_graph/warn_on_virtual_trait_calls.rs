//@ print-targets
//@ stdout
//@ stderr

trait Trait {
    fn method(&self);
}

fn impl_method() {}
struct S;
impl Trait for S {
    fn method(&self) {
        impl_method();
    }
}

fn make_virtual_trait_call(v: Box<dyn Trait>) {
    v.method();
}

#[test]
fn test() {
    make_virtual_trait_call(Box::new(S));

    let v: &dyn Trait = &S;
    v.method();
}
