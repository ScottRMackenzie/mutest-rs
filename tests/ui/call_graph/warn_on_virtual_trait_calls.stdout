tests -(0)-> make_virtual_trait_call at tests/ui/call_graph/warn_on_virtual_trait_calls.rs:17:1: 17:46 (#0)
  (0) test

targets: 1 total; 1 safe; 0 unsafe (0 tainted)
